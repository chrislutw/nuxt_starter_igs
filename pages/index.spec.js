import { shallowMount } from '@vue/test-utils'
import index from '~/pages/index'
import NuxtLink from '~/.nuxt/components/nuxt-link'
describe('index', () => {
  const factory = () =>
    shallowMount(index, {
      mocks: {
        localePath: () => {}
      }
    })
  beforeAll(() => {
    index.components.NuxtLink = NuxtLink
  })

  test('mounts properly', () => {
    const wrapper = factory()
    wrapper.vm.localePath = () => {}
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
