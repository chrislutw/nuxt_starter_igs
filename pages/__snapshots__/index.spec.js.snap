// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `
<section class="container">
  <div>
    <logo-stub></logo-stub>
    <h1 class="title">test1</h1>
    <h2 class="subtitle">My hunky-dory Nuxt.js project</h2>
    <h2 class="subtitle">
      <nuxt-link-stub>Login</nuxt-link-stub>
    </h2>
    <div class="links"><a href="https://nuxtjs.org/" target="_blank" class="button--green">Documentation</a> <a href="https://github.com/nuxt/nuxt.js" target="_blank" class="button--grey">GitHub</a></div>
  </div>
</section>
`;
